docs/design/2020-06-24-placement-rules-in-sql.md

```diff
@@ -0,0 +1,391 @@
+	ADD PLACEMENT LABEL="-zone=sh" ROLE=follower COUNT=2;
+```
+
+This statement indicates PD to schedule the leader to `sh`, and do not schedule followers to `sh`.
+
+`ADD PLACEMENT` can be used to separate data on different data centers, which may help to reduce multi-region latency.
+
+Adding placement can be done through adding a placement rule in PD. The statement must wait until the PD returns a message.
+
+### Altering placement
+
+Altering current placement rules can be done by one or more `ALTER PLACEMENT` clauses:
+
+```sql
+ALTER TABLE table_name
+	ALTER PLACEMENT LABEL=label ROLE=role COUNT=count,
+	...
+```
+
+This statement indicates TiDB to overwrite the current placement rule with the same `role`. It affects all data, including indexes, of table `table_name`.
+
+This kind of operation can be used in adding more replicas for important data, or adding a TiFlash replica.
+
+Altering placement can be done through override the current placement rule by a new one.
+
+Note that when the rules in the statement conflict, an error should occur. For example, an error should be reported when executing this statement:
+
+```sql
+ALTER TABLE test
+	ALTER PLACEMENT LABEL="+zone=sh" ROLE=voter COUNT=3,
+	ALTER PLACEMENT LABEL="+zone=bj" ROLE=voter COUNT=3;
+```
+
+`+zone=sh` conflicts with `+zone=bj`, because they’re defining the same role `voter`. These two constraints can not be satisfied both.
+
+The statement must wait until the PD returns a message.
+
+### Label configuration
+
+`LABEL` in the statement indicates the label constraints. Data is placed on the TiKV nodes whose labels conform to `LABEL` constraint. `LABEL` must be specified in `ADD PLACEMENT` or `ALTER PLACEMENT` clauses.
+
+Parameter `label` should be a string and in such a format: `{+|-}key=value,...`. Prefix `+` indicates that data can only be placed on those TiKV nodes whose labels contain such labels, and `-` indicates that data can’t be placed on those TiKV nodes whose labels contain such labels. `key` here refers to the label name, and `value` is the label value. `key` should be defined in TiKV labels.
+
+For example, `+zone=sh,+zone=bj` indicates to place data only in `sh` and `bj` zone. Also, `zone` should have already be defined in TiKV configuration. For example:
+
+```sql
+[server]
+labels = "zone=<zone>,rack=<rack>,host=<host>"
+```
+
+Label constraints can be implemented by defining `label_constraints` field in placement rules. `+` and `-` correspond to property `op`. Specifically, `+` is equivalent to `In` and `-` is equivalent to `NotIn`.
+
+For example, `+zone=sh,+zone=bj` is equivalent to:
+
+```
+"label_constraints": [
+	{"key": "zone", "op": "in", "values": ["sh", “bj”]}
+]
+```
+
+### Role configuration
+
+`ROLE` in the statement defines the role of replicas. Newly added replicas are in such roles.
+
+`ROLE` is equivalent to property `role` in placement rules. it can be one of such values: leader, follower, learner, voter.
```

Comment:
Would TiDB return the available values when an unrecognized value is received?

Reply:
@xhebox Please notice this in the implementation, it's a usability improvement.